src/event.rs

```diff
@@ -256,19 +257,19 @@ impl<T: Clone> Clone for Event<'static, T> {
 }
 
 impl<'a, T> Event<'a, T> {
-    pub fn map_nonuser_event<U>(self) -> Result<Event<'a, U>, Event<'a, T>> {
+    pub fn map_nonuser_event<U>(self) -> Option<Event<'a, U>> {
```

Comment:
I did this because clippy didn't like it ([`result_large_err`](https://rust-lang.github.io/rust-clippy/master/#result_large_err)) and I didn't realize it was a public API. @maroider am I correct that you intended to revert this? In any case this can be reverted but a clippy `allow` will be needed.

Reply:
Yeah, I intended to revert it, but then I forgot